### PR TITLE
Copyediting the rest of the code

### PR DESCRIPTION
**Pull request by austinzheng**
@jpsim @tgoyne @bdash

NFC

- Copyedited all remaining ObjC files
- Copyedited all Swift files
- Additional style and correctness fixes

### CODE REVIEW

File: Realm/RLMProperty.h [hunk 1]
```diff
@@ -42,68 +42,67 @@ NS_ASSUME_NONNULL_BEGIN
...
+ Returns a Boolean value that indicates whether a given property object is equal to the receiver.
  */
 - (BOOL)isEqualToProperty:(RLMProperty *)property;
 
 @end
 
 
 /**
- This class describes a specific property on a given class.
+ A `RLMPropertyDescriptor` instance represents a specific property on a given class.
```
Comment (tgoyne) [inline Realm/RLMProperty.h]: An

File: Realm/RLMResults.h [hunk 2]
```diff
@@ -285,7 +285,7 @@ NS_ASSUME_NONNULL_BEGIN
 - (NSNumber *)sumOfProperty:(NSString *)property;
 
 /**
- Returns the average value of a given property for all the objects represented by
+ Returns the average value of a given property for over the objects represented by
```
Comment (tgoyne) [inline Realm/RLMResults.h]: Changed the wrong word here.

File: RealmSwift-swift2.2/List.swift [hunk 3]
```diff
@@ -133,173 +128,175 @@ public final class List<T: Object>: ListBase {
...
+     Returns a sorted `Results` from the list.
 
-    - parameter property:  The property name to sort by.
-    - parameter ascending: The direction to sort by.
+     - parameter property:  The property name to sort by.
+     - parameter ascending: The direction to sort by.
 
-    - returns: `Results` containing elements sorted by the given property.
-    */
+     - returns: An `Results` containing the list objects sorted by the specified property.
```
Comment (tgoyne) [inline RealmSwift-swift2.2/List.swift]: A Results (there's a bunch of these; not marking the rest).

File: Realm/RLMRealm.h [hunk 4]
```diff
@@ -345,12 +346,12 @@ typedef void (^RLMNotificationBlock)(NSString *notification, RLMRealm *realm);
...
  Once added, this object is considered to be managed by the Realm. It can be retrieved
  using the `objectsWhere:` selectors on `RLMRealm` and on subclasses of `RLMObject`.
+
  When added, all child relationships referenced by this object will also be added to 
  the Realm if they are not already in it.
  
  If the object or any related objects are already being managed by a different Realm
- an exception will be thrown. Use
- `-[RLMObject createInRealm:withObject]` to insert a copy of a persisted object
+ an exception will be thrown. Use `-[RLMObject createInRealm:withObject]` to insert a copy of a managed object
```
Comment (jpsim) [inline Realm/RLMRealm.h]: missing colon: `-[RLMObject createInRealm:withObject:]`

File: RealmSwift-swift2.2/LinkingObjects.swift [hunk 5]
```diff
@@ -133,7 +134,7 @@ public final class LinkingObjects<T: Object>: LinkingObjectsBase {
      Returns the index of the first object matching the given predicate,
      or `nil` if no objects match.
 
-     - parameter predicate: The predicate to filter the objects.
+     - parameter predicate: The predicate to use to filter the objects.
```
Comment (jpsim) [inline RealmSwift-swift2.2/LinkingObjects.swift]: Maybe this instead?

> The predicate with which to filter the objects.

File: RealmSwift-swift2.2/LinkingObjects.swift [hunk 6]
```diff
@@ -170,44 +171,43 @@ public final class LinkingObjects<T: Object>: LinkingObjectsBase {
...
-       collection's objects.
+     - returns: An `Array` containing the results.
      */
     public override func valueForKeyPath(keyPath: String) -> AnyObject? {
         return rlmResults.valueForKeyPath(keyPath)
     }
 
     /**
-     Invokes `setValue(_:forKey:)` on each of the collection's objects using the specified value and key.
+     Invokes `setValue(_:forKey:)` on each of the linking object collection's objects using the specified `value` and
```
Comment (jpsim) [inline RealmSwift-swift2.2/LinkingObjects.swift]: "linking objects collection"

File: RealmSwift-swift2.2/ObjectSchema.swift [hunk 7]
```diff
@@ -20,36 +20,40 @@ import Foundation
...
 
     // MARK: Properties
 
     internal let rlmObjectSchema: RLMObjectSchema
 
-    /// Array of persisted `Property` objects for an object.
+    /**
+     An array of `Property` instances representing the persisted properties of a class described by the schema.
+
+     -see `Property`
```
Comment (jpsim) [inline RealmSwift-swift2.2/ObjectSchema.swift]: `- see:`

File: RealmSwift-swift2.2/RealmConfiguration.swift [hunk 8]
```diff
@@ -22,21 +22,23 @@ import Realm.Private
...
-    create a `Realm` instance.
-
-    `Realm.Configuration` instances are just plain Swift structs, and unlike
-    `Realm` and `Object`s can be freely shared between threads. Creating
-    configuration objects for class subsets (by setting the `objectTypes`
-    property) can be expensive, and so you will normally want to cache and reuse
-    a single configuration object for each distinct configuration that you are
-    using rather than creating a new one each time you open a `Realm`.
-    */
+     An `Configuration` instance describes the different options used to
```
Comment (jpsim) [inline RealmSwift-swift2.2/RealmConfiguration.swift]: "a `Configuration`"

**Comment by jpsim** [comment]
👍

**Pull request merged**